classes/plugininfo/factor.php

```diff
@@ -239,5 +239,21 @@ public function load_settings(\part_of_admin_tree $adminroot, $parentnodename, $
 
         $adminroot->add($parentnodename, $settings);
     }
-}
 
+    /**
+     * Checks that given factor exists.
+     *
+     * @param string $factorname
+     *
+     * @return bool
+     */
+    public static function factor_exists($factorname) {
```

Comment:
Seems like some conceptual cross over with get_factor, wondering if we can merge these, or just call get_factor from here?